src/seedu/addressbook/data/person/Name.java

```diff
@@ -60,4 +60,10 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
+    @Override
+    public int compareTo(Name other) {
+        if (this.equals(other))
+            return 0;
+        return this.fullName.compareTo(other.fullName);
```

Comment:
Hmm have you tried removing the `if` conditional and see if it works? Since `equals` has been overridden for `Name`, I suspect `compareTo` might be able to work as expected even without the `if` conditional that you have added here.